src/sc_containers.c

```diff
@@ -1631,6 +1637,11 @@ sc_hash_array_lookup (sc_hash_array_t * hash_array, void *v, size_t *position)
   int                 found;
   void              **found_void;
 
+  /* verify general invariant */
+  SC_ASSERT (hash_array != NULL);
+  SC_ASSERT (hash_array->a.elem_count == hash_array->h->elem_count);
+  SC_ASSERT (hash_array->internal_data->foreach_fn == NULL);
```

Comment:
Would it make sense to check for `hash_array->internal_data->v == NULL` here too, since it is set to NULL after every usage, similar to `foreach_fn`?

Reply:
Thanks for pushing the addition!